python/ray/data/tests/test_execution_optimizer.py

```diff
@@ -607,6 +607,33 @@ def __call__(self, x):
     assert isinstance(physical_op.input_dependencies[0], InputDataBuffer)
 
 
+def test_read_map_batches_operator_fusion_with_random_shuffle_operator(
```

Comment:
can we add more test cases, e.g. for multiple `map_batches` and multiple `random_shuffle`? Let's also check negative case - all other all-to-all operators except `random_shuffle` should not be fused.